cmd/fs-v1-helpers.go

```diff
@@ -344,6 +344,59 @@ func fsCreateFile(ctx context.Context, filePath string, reader io.Reader, falloc
 	return bytesWritten, nil
 }
 
+func panfsCreateFile(ctx context.Context, filePath string, reader io.Reader, fallocSize int64, objMode os.FileMode, ownerID, groupID int) (int64, error) {
+	if filePath == "" || reader == nil {
+		logger.LogIf(ctx, errInvalidArgument)
+		return 0, errInvalidArgument
+	}
+
+	if err := checkPathLength(filePath); err != nil {
+		logger.LogIf(ctx, err)
+		return 0, err
+	}
+
+	// This mode bits wasn't changed to param, because files create in temp folder
```

Comment:
BTW, as now we own our version of this function, I'd go away from 777 rights on temporary directories in S3 to something lilke 755 or even 700.

And if this function is designed to create file in temporary folder, I'd call it something like "panfsCreateTemporaryFile"

Reply:
Outdated, there is no more our own version